pgaf/ingest.py

```diff
@@ -3,29 +3,99 @@
 from sqlalchemy import create_engine, select
 from sqlalchemy.orm import Session
 
-import sqla_schema as sch
+import json
+import pandas as pd
+from datetime import date,datetime,timedelta
 
+import sqla_schema as sch
 
 manifest = '/allen/scratch/aindtemp/david.feng/epc/manifest.json'
 cache = EcephysProjectCache(manifest=manifest)
 
-sessions = cache.get_session_table()
-cache.get_probes()
-cache.get_channels()
-cache.get_units()
+sessions = cache.get_session_table(suppress=[])
+probes = cache.get_probes()
+channels = cache.get_channels()
+units = cache.get_units()
 
 for idx,row in sessions.head(3).iterrows():
     cache.get_session_data(idx)
 
+with open('SECRETS','r') as f:
+    conn_info = json.load(f)
+
+conn_string = f"postgresql://{conn_info['user']}:{conn_info['password']}@{conn_info['host']}:5432/{conn_info['dbname']}"
 
-engine = create_engine('sqlite://')
+engine = create_engine(conn_string)
+sch.Base.metadata.drop_all(engine)
 sch.Base.metadata.create_all(engine)
 
-sessions.to_sql('session', engine, if_exists='replace')
+# preprocess sessions
+sessions['acquisition_datetime'] = sessions['date_of_acquisition'].map(lambda v: datetime.strptime(v, "%Y-%m-%dT%H:%M:%S%z"))
+sessions['publication_datetime'] = sessions['published_at'].map(lambda v: datetime.strptime(v, "%Y-%m-%dT%H:%M:%S%z"))
+sessions.drop(columns=['date_of_acquisition',
+                       'published_at'],
+              inplace=True)
+
+# ingest genotypes
+genotypes = pd.DataFrame(data={'name': sessions['full_genotype'].unique()})
+genotypes.to_sql('genotype', engine, index_label='id', if_exists='append')
+
+# ingest mice
+mice = sessions[['acquisition_datetime', 'age_in_days', 'specimen_id','sex','full_genotype']]
+mice['date_of_birth'] = mice.apply(lambda r: r['acquisition_datetime'].date() - timedelta(r['age_in_days']), axis=1)
+mice = mice.merge(genotypes.reset_index(), left_on='full_genotype', right_on='name', how='left').rename(columns={'index':'genotype_id'})
+mice = mice[['specimen_id','sex','date_of_birth','genotype_id']].set_index('specimen_id')
+mice.index.name ='id'
+mice.to_sql('mouse', engine, index_label='id', if_exists='append')
+
+# ingest structures
```

Comment:
@yonibrowning this is how I got a complete structures table.  download from the commented URL.

Reply:
Got it! Thanks.